python/ray/serve/_private/deployment_state.py

```diff
@@ -519,14 +524,18 @@ def check_ready(self) -> ReplicaStartupStatus:
                 self._pid, self._actor_id, self._node_id, self._node_ip = ray.get(
                     self._allocated_obj_ref
                 )
-            except Exception:
+            except Exception as e:
                 logger.exception(
                     f"Exception in replica '{self._replica_tag}', "
                     "the replica will be stopped."
                 )
-                return ReplicaStartupStatus.FAILED
+                # NOTE(zcin): we should use str(e) instead of traceback.format_exc()
+                # here because the exception may be a RayTaskError, in which case the
+                # full details of the error is not displayed properly with
+                # traceback.format_exc().
+                return ReplicaStartupStatus.FAILED, str(e)
```

Comment:
I think we might want to use `repr(e)` instead of `str(e)`, but there's likely no difference in practice with common exception types

Reply:
For RayTaskErrors specifically, this seems to cause weird formatting:

<img width="611" alt="Screen Shot 2023-05-01 at 5 59 42 PM" src="https://user-images.githubusercontent.com/15851518/235557335-b3353606-5c04-492f-9baf-a65a8702a6ef.png">

Reply:
Got it. Btw, can we unwrap the `RayTaskError` so we only show the user their underlying exception instead? This can be done with `e.as_instanceof_cause()`